### PR TITLE
Can we build the keyboard using grid?

### PR DESCRIPTION
**Pull request by JimFung**
tl;dr: I don't think we could.

image

So....initially I wanted to change the html such that there would be a `row` div which housed all the keys. Kind of something like this.
```html
<div class='keyboard'>
   <div class='row'>
      top row of keys
   </div>
   <div class='row'>
      spacer
      middle row of keys
      spacer
   </div>
   <div class='row'>
      spacer
      bottom row of keys
      spacer
   </div>
</div>
```

But having looked at the code I don't think that's possible without rewriting some pretty important pieces of the game logic, which expect keyboard being an array of `{ type, label, state }` objects. Instead of doing that (cause I didn't want to read through all this js) or creating a new array in the shape I want purely for rendering purposes (not an awful idea but keeping state between these 2 different keyboard arrays in sync seems like a problem best avoided) I wanted to see if we could do this purely with css grid. tldr; we can't because grids want you to have an even number of columns for each row (cause y'know its a grid). You can _kind_ of fake it... I got each row to show the correct letters but getting things to be centered horizontally won't work because you'd have to render keys over grid lines, which css grid doesn't let you do.

### CODE REVIEW

File: public/input.css [hunk 1]
```diff
@@ -88,12 +88,28 @@ a {
...
-  justify-content: center;
-  margin: 0 auto 8px;
-  font-size: 1.5rem;
+  display: grid;
+  grid-template-columns: repeat(12, 1fr);
+  gap: 0.25rem;
+}
+
+/* Kind of redundant, here for completeness */
+.keyboard > button:nth-child(-1n + 12) {
```
Comment (JimFung) [inline public/input.css:97]: Docs on nth-child functional notation: https://developer.mozilla.org/en-US/docs/Web/CSS/:nth-child#functional_notation

tl;dr - This tells the first 12 elements they should render on the first row. 

Note: because CSS prefers whichever styles are written last the order of these classes  matters here.

**Comment by wachunga** [comment]
Based on the screenshot, I think you're out of date. We actually were using `display: grid` at some point but we ended up with defined rows to accomplish a proper qwerty layout

**Comment by JimFung** [comment]
Can you clarify what you mean by qwerty layout?

I should have been clearer about what I was trying to achieve (my bad 😅). I'm trying to make the keyboard look more like the keyboard found on wordle, that is with the keys laid out in an upside down trapezoid.

**Comment by JimFung** [comment]
Ah I see what you mean now @wachunga I am in fact out of date.
